src/global/pidfile.cc

```diff
@@ -170,9 +170,15 @@ int pidfh::open(const ConfigProxy& conf)
     .l_len = 0
   };
   int r = ::fcntl(pf_fd, F_SETLK, &l);
-  if (r < 0) {
-    derr << __func__ << ": failed to lock pidfile "
-	 << pf_path << " because another process locked it." << dendl;
+  if (r != 0) {
```

Comment:
i'd suggest use `cpp_strerror()` to translate the errno to human-readable error message.